### PR TITLE
Fix fire direction

### PR DESCRIPTION
**Pull request by frozenbanana**
## Purpose
  - Make projectiles looks like the projectiles are coming from the weapon

## Changes
 - `OffsetComponent` in `LevelObjectCreator.cpp` modified to have an offset in x-axsis. This make the projectile start from something that looks like the gun.
 - `postition` in `WeaponSystem.cpp` weapon calculation to take in account offset in x-axis.
 - `translate(..)` in `FirstPersonWeaponRenderer.cpp` to not have weapon be covered by HUD

## Testing
How have one tested the feature to ensure it works?
- [x] Tested in Sandbox
- [x] Tested in Multiplayer with 2 clients
- [ ] Tested in Multiplayer with more than 2 clients
- [ ] Tested in Benchmark

## Context 
I had issues getting the right direction in certain directions. With the help of @Ceanze we got to work by including xoffset by multiplying with players GetRight instead of direct quartnions multiplication.

## Screen Captures
image

### CODE REVIEW

File: CrazyCanvas/Include/ECS/Systems/Player/WeaponSystem.h [hunk 1]
```diff
@@ -106,6 +106,6 @@ class WeaponSystem : public LambdaEngine::System
 	static LambdaEngine::TUniquePtr<WeaponSystem> s_Instance;
 
 private:
-	float	m_ZeroDist	= 5.0f;
-	float	m_YAngle	= 0.0f;
+	float	m_ZeroDist	= 8.0f;
+	float	m_YAngle	= 5.f;
```
Comment (TimTheToad) [inline CrazyCanvas/Include/ECS/Systems/Player/WeaponSystem.h]: defining convention, either 5.0f or 8.f ^^

**Review by TimTheToad — CHANGES_REQUESTED** [review]
nice work!

File: CrazyCanvas/Include/ECS/Systems/Player/WeaponSystem.h (hunk 1 above)
Comment (frozenbanana) [reply CrazyCanvas/Include/ECS/Systems/Player/WeaponSystem.h]: you are right! fixed

**Review by TimTheToad — APPROVED** [review]
dutti!

**Pull request merged**